### PR TITLE
[JS Package]Connection: add manage connection dialog component

### PR DESCRIPTION
**Pull request by nunyvega**
Completes the first task of #26577

#### Changes proposed in this Pull Request:

- Create the manage connection dialog component's skeleton and story
Screen Shot 2022-10-06 at 16 55 33

Note: The component will not be used yet. it will be styled in a follow-up PR.
You'll find the detailed steps [here](https://github.com/Automattic/jetpack/issues/26577).

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:

* Build and open the storybook
* Look for the component in `JS PACKAGES / Connection / Manage Connection Dialog`
* Confirm that the component is there, and that both controls are available (isOpen, title)

### CODE REVIEW

File: projects/js-packages/connection/components/manage-connection-dialog/index.jsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+				<Modal
+					title={ title }
+					contentLabel={ title }
+					aria={ {
+						labelledby: 'jp-connection__manage-dialog__heading',
+					} }
+					shouldCloseOnClickOutside={ false }
+					shouldCloseOnEsc={ false }
+					isDismissible={ false }
+					className={ 'jp-connection__manage-dialog' }
```
Comment (retrofox) [inline projects/js-packages/connection/components/manage-connection-dialog/index.jsx:29]: Are we getting styles when applying this CSS class? And beyond of this, I think we should use CSS modules.

File: projects/js-packages/connection/components/manage-connection-dialog/stories/index.jsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export default {
+	title: 'JS Packages/Connection/Manage Connection Dialog',
+	component: ManageConnectionDialog,
+};
+
+const Template = args => <ManageConnectionDialog { ...args } />;
+
+export const _default = Template.bind( {} );
+_default.args = {
+	isOpen: true,
```
Comment (retrofox) [inline projects/js-packages/connection/components/manage-connection-dialog/stories/index.jsx:13]: You could set the default value for the props. `title` in this case.

```suggestion
	isOpen: true,
	title: 'Manage your Jetpack connection',
```

Screen Shot 2022-10-06 at 12 26 05

File: projects/js-packages/connection/components/manage-connection-dialog/index.jsx (hunk 1 above)
Comment (nunyvega) [reply projects/js-packages/connection/components/manage-connection-dialog/index.jsx:29]: Nothing yet, the component is not yet styled. I plan to tackle the styling in a different PR once I have the functionality

Comment (nunyvega) [reply projects/js-packages/connection/components/manage-connection-dialog/index.jsx:29]: Would you recommend removing the class while unused?

**Review by retrofox — DISMISSED** [review]

**Review by retrofox — APPROVED** [review]
LGTM

**Pull request merged**